_posts/2017-11-07-haskell-learning-group-our-progress-and-updates.md

```diff
@@ -0,0 +1,71 @@
+
+[ramdajs]: http://ramdajs.com
+
+## Types
+
+We went through many topics in the book and since the last update we've covered chapter four, five and discussed a half of the chapter six which covers Typeclasses. The main topics were:
+* Basic DataTypes
+* Types
+* Typeclasses
+
+We learned important basics that helps a lot when you read and try to understand what certain code does. For instance understanding how to read type signature, so something like this starting to make sense:
+
+`(Num a, Num b) => a -> b -> b`
+
+In this case there are two constraints, one for each parameter. They must be of a typeclass Num. You might already have some idea what function might do, based on its type signature.
+We had a look on different examples of how type signatures limit a scope of what body of the function might be. Even more, some  type signatures of function tells us exactly what it does because its the only possible implementation of the function. The simplest example of that will be an identity function:
+
+```haskell
+id :: a -> a
+id x = x
+```
+
+The only thing this function can possibly do is return parameter that was passed to it. More advanced example when there is only one possible implementation is function composition:
+
+```haskell
+f :: (b -> c) -> (a -> b) -> (a -> c)
+f x y = \z -> x(y(z))
+```
+
+It's not obvious from the first look at this type definition that there is no other possible implementation of a function based on its type. Compiler won't let us implement it in any other way. 
+
+## Katas and Exercises
+
+I like the idea of projects and platforms like [exercism.io][exercism], [Project Euler][project-euler] and similar. There are small well defined problems, that can be solved in multiple ways. People share solutions, comment other people's code and try to improve themselves. Its can be very handy for those who learning a new programming language. In case of Haskell, you can see how with multiple iteration you improve the initial solution from having i.e. multiple if-else to use pattern matching and so on. 
+I've started a [Haskell Exercises Repo][haskell-exercises-repo], which is basically nothing more than a collection of such problems and solutions to them. Most of them wouldn't make sense to make as a separate repository because of small size. So each directory is a separate exercise. It can be just one file or a Stack Project with tests and code that make them green. This repository has been created for learning purposes and accept contributions either as improvements to existing solutions or as new solutions. 
+
+[haskell-exercises-repo]: https://github.com/HeeL/haskell-exercises
+[exercism]: http://exercism.io
+[project-euler]: https://projecteuler.net
+
+## Haskell Meetup @ HolidayCheck
```

Comment:
This is awesome info! IMO it also would be handy to make this info as a poster and put it directly on the homepage or something. Because there is a chance that someone may not come across this one since this person isn't possibly interested in group meetings overall.

Reply:
yeah, I would also be interested in some analytics numbers from our blog. For instance how many people from Munich actually visiting our blog regularly.

Reply:
yeah, and I just taking about some additional info of such kind - as some print poster, or banner on the tech blog etc. - Let's spread the word out about this endeavour! :)

Reply:
We are often hosting all different kinds of MeetUps and workshops (i.e. Nix, JavaScript, Agile and so on). Maybe we create an "Events @ HC" section here and promote everything that happens at HolidayCheck?

Reply:
would be super nice - For now I am lost with what happens when. Good idea @HeeL